compiler/src/iree/compiler/Codegen/ExternalInterfaces/Utils.h

```diff
@@ -55,9 +55,15 @@ Value calculateStorageSizeInBytesImpl(Attribute attr, Location loc,
                                       ValueRange dynamicDims);
 
 /// Returns a dictionary attribute that contains the materialized encoding info,
-/// i.e., serialized MaterializeEncodingInfo struct.
+/// i.e., serialized MaterializeEncodingInfo struct. The EncodingAttr attribute
+/// is attached to the dictionary, if it is present in `type` and
+/// `addEncodingAttr` is true.
+/// The `addEncodingAttr` is mainly for VMVX ukernel path because the ukernel
+/// ops lowering requires all the information. There are no direct mappings from
+/// layouts to ukernels.
```

Comment:
Can't we update the `VMVXEncodingLayoutAttr` to include all the necessary information instead (possibly as a follow-up)?